### PR TITLE
More Search page improvements

### PR DESCRIPTION
**Pull request by Mikunj**
**Changes:**

- Hooked up Streaming, Seasons and Categories to search results.
- Improved `ResultList` layout.
    - Grid will now adjust item size and margins dynamically so that the spacing between items remains tight and even.
- The back arrow now shows when media pages are loading. The issue before was that if you accidentally tapped a media, you had to wait for the whole thing to load (takes a long time) before you could go back. Showing the back button here makes it so that users can go back if they accidentally tapped a media they did not want.

**TODO**

- Swap out filter view modal popup with the ones used in feed (when you press the 3 dot menu)
- Fix up filters
   - When you view Top list, then pull to refresh, it will give you completely different results.
- Minor UX stuff.

### CODE REVIEW

File: src/common/utils/index.js [hunk 1]
```diff
@@ -3,3 +3,103 @@ export const idExtractor = item => item.id;
...
+
+/**
+ * Get the Best Grid Spacing from the given itemWidths.
+ * The best spacing is the one with a margin that is close to minMargin.
+ *
+ * @param {Array} itemWidths An array of widths. I.e [10, 20, 30]
+ * @param {Int} availableWidth The width of the grid/The amount of space we can work with.
+ * @param {Int} minMargin The minimum margin to keep between items and around the edges.
+ * @returns {Object} An object containing the best spacing and the item width associated with it.
+ */
```
Comment (edencakir) [inline src/common/utils/index.js]: 💯 we should have these everywhere ™️ . It's like browsing Android's core classes

File: src/screens/Search/FilterSub.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ class FilterSub extends Component {
   static navigationOptions = ({ navigation }) => ({
     title: navigation.state.params.title || navigation.state.params.label,
     headerLeft: (
-      <Button transparent color="white" onPress={navigation.goBack}>
+      <Button transparent color="white" onPress={() => navigation.goBack()}>
```
Comment (edencakir) [inline src/screens/Search/FilterSub.js]: `onPress={navigation.goBack}` was it passing `null`?

File: src/screens/Search/Lists/ResultsList.js [hunk 3]
```diff
@@ -27,60 +29,91 @@ const styles = StyleSheet.create({
...
+        )}
+      </TouchableOpacity>
+    </View>
   );
 };
 
+ResultsList.propTypes = {
+  hits: PropTypes.array.isRequired,
+  onPress: PropTypes.func.isRequired,
+  style: PropTypes.object,
```
Comment (edencakir) [inline src/screens/Search/Lists/ResultsList.js]: Feel free to use `ViewPropTypes` for the style object.

File: src/screens/Search/FilterSub.js (hunk 2 above)
Comment (Mikunj) [reply src/screens/Search/FilterSub.js]: Don't know but when i tried tapping the back button on the view i couldn't go back, so i'm assuming so.

**Pull request merged**